sdk/spring/azure-spring-boot/src/main/java/com/azure/spring/autoconfigure/aad/AADAuthenticationProperties.java

```diff
@@ -139,23 +141,55 @@
          * Expected UserGroups that an authority will be granted to if found in the response from the MemeberOf Graph
          * API Call.
          */
-        private List<String> allowedGroups = new ArrayList<>();
+        private List<String> allowedGroupNames = new ArrayList<>();
 
+        private Set<String> allowedGroupIds = new HashSet<>();
+
+        public Set<String> getAllowedGroupIds() {
+            return allowedGroupIds;
+        }
+
+        public void setAllowedGroupIds(Set<String> allowedGroupIds) {
+            this.allowedGroupIds = allowedGroupIds;
+        }
+
+        public List<String> getAllowedGroupNames() {
+            return allowedGroupNames;
+        }
+
+        public void setAllowedGroupNames(List<String> allowedGroupNames) {
+            this.allowedGroupNames = allowedGroupNames;
+        }
+
+        @DeprecatedConfigurationProperty(
+            reason = "In order to distinguish between allowed-group-ids and allowed-group-names, set allowed-groups "
+                + "deprecated.",
+            replacement = "azure.activedirectory.user-group.allowed-group-names")
         public List<String> getAllowedGroups() {
-            return allowedGroups;
+            return allowedGroupNames;
         }
 
+        @DeprecatedConfigurationProperty(
```

Comment:
According to the [javadoc](https://github.com/spring-projects/spring-boot/blob/47516b50c39bd6ea924a1f6720ce6d4a71088651/spring-boot-project/spring-boot-tools/spring-boot-configuration-processor/src/test/java/org/springframework/boot/configurationsample/DeprecatedConfigurationProperty.java#L31), `@DeprecatedConfigurationProperty` must be used on the getter of the deprecated element.

Reply:
Please refer to this sample https://github.com/spring-projects/spring-boot/blob/47516b50c39bd6ea924a1f6720ce6d4a71088651/spring-boot-project/spring-boot-tools/spring-boot-configuration-processor/src/test/java/org/springframework/boot/configurationsample/simple/DeprecatedSingleProperty.java#L27.